llm_bench/python/utils/model_utils.py

```diff
@@ -95,13 +90,31 @@ def analyze_args(args):
     return model_path, model_framework, model_args, model_name
 
 
-def get_use_case(model_name_list):
-    for model_name in reversed(model_name_list):
+def get_use_case(model_name_or_path):
+    # 1. try to get use_case from model name
+    path = os.path.normpath(model_name_or_path)
+    model_names = path.split(os.sep)
+    for model_name in reversed(model_names):
         for case, model_ids in USE_CASES.items():
             for model_id in model_ids:
                 if model_name.lower().startswith(model_id):
-                    log.info(f'==SUCCESS FOUND==: use_case: {case}, model_name: {model_name}')
+                    log.info(f'==SUCCESS FOUND==: use_case: {case}, model_type: {model_name}')
                     return case, model_name
+
+    # 2. try to get use_case from model config
```

Comment:
Shall we take this as first choice?

Reply:
I think we can consider that later, there is some possible issues with model versioning, if they stored with remote code and share the same model_type (e.g. we have different classes for chatglm and chatglm2/3, model_type in config the same